rust/src/lib/ifaces/bond.rs

```diff
@@ -0,0 +1,475 @@
+    #[serde(alias = "0")]
+    Dropped,
+    #[serde(alias = "1")]
+    Delivered,
+}
+
+impl BondAllPortsActive {
+    pub fn to_u8(&self) -> u8 {
+        match self {
+            Self::Dropped => 0,
+            Self::Delivered => 1,
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename_all = "lowercase")]
+pub enum BondArpAllTargets {
+    #[serde(alias = "0")]
+    Any,
+    #[serde(alias = "1")]
+    All,
+}
+
+impl BondArpAllTargets {
+    pub fn to_u32(&self) -> u32 {
+        match self {
+            Self::Any => 0,
+            Self::All => 1,
+        }
+    }
+}
+
+const BOND_STATE_ACTIVE: u8 = 0;
+const BOND_STATE_BACKUP: u8 = 1;
+
+const BOND_ARP_VALIDATE_NONE: u32 = 0;
+const BOND_ARP_VALIDATE_ACTIVE: u32 = 1 << BOND_STATE_ACTIVE as u32;
+const BOND_ARP_VALIDATE_BACKUP: u32 = 1 << BOND_STATE_BACKUP as u32;
+const BOND_ARP_VALIDATE_ALL: u32 =
+    BOND_ARP_VALIDATE_ACTIVE | BOND_ARP_VALIDATE_BACKUP;
+const BOND_ARP_FILTER: u32 = BOND_ARP_VALIDATE_ALL + 1;
+const BOND_ARP_FILTER_ACTIVE: u32 = BOND_ARP_VALIDATE_ACTIVE | BOND_ARP_FILTER;
+const BOND_ARP_FILTER_BACKUP: u32 = BOND_ARP_VALIDATE_BACKUP | BOND_ARP_FILTER;
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename_all = "lowercase")]
+pub enum BondArpValidate {
+    None,
+    Active,
+    Backup,
+    All,
+    Filter,
+    #[serde(rename = "filter_active")]
+    FilterActive,
+    #[serde(rename = "filter_backup")]
+    FilterBackup,
+}
+
+impl BondArpValidate {
+    pub fn to_u32(&self) -> u32 {
+        match self {
+            Self::None => BOND_ARP_VALIDATE_NONE,
+            Self::Active => BOND_ARP_VALIDATE_ACTIVE,
+            Self::Backup => BOND_ARP_VALIDATE_BACKUP,
+            Self::All => BOND_ARP_VALIDATE_ALL,
+            Self::Filter => BOND_ARP_FILTER,
+            Self::FilterActive => BOND_ARP_FILTER_ACTIVE,
+            Self::FilterBackup => BOND_ARP_FILTER_BACKUP,
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename_all = "lowercase")]
+pub enum BondFailOverMac {
+    #[serde(alias = "0")]
+    None,
+    #[serde(alias = "1")]
+    Active,
+    #[serde(alias = "2")]
+    Follow,
+}
+
+impl BondFailOverMac {
+    pub fn to_u8(&self) -> u8 {
+        match self {
+            Self::None => 0,
+            Self::Active => 1,
+            Self::Follow => 2,
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename_all = "lowercase")]
+pub enum BondPrimaryReselect {
+    #[serde(alias = "0")]
+    Always,
+    #[serde(alias = "1")]
+    Better,
+    #[serde(alias = "2")]
+    Failure,
+}
+
+impl BondPrimaryReselect {
+    pub fn to_u8(&self) -> u8 {
+        match self {
+            Self::Always => 0,
+            Self::Better => 1,
+            Self::Failure => 2,
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+#[serde(rename_all = "lowercase")]
```

Comment:
Every member has a `serde(rename)`. So there is no need for this `rename_all()`.

Reply:
Thanks!